packages/labs/task/src/task.ts

```diff
@@ -107,26 +127,37 @@ export class Task<T extends [...unknown[]] = any, R = any> {
 
   constructor(
     host: ReactiveControllerHost,
-    task: TaskFunction<T, R>,
-    getDependencies: DepsFunction<T>
+    task: TaskFunction<T, R> | TaskConfig<T, R>,
+    args?: ArgsFunction<T>
   ) {
     this._host = host;
     this._host.addController(this);
+    if (typeof task === 'object') {
+      const taskConfig = task as TaskConfig<T, R>;
+      args = taskConfig.args;
+      task = taskConfig.task;
+      if (taskConfig.canRun !== undefined) {
+        this._canRun = taskConfig.canRun;
+      }
+    }
     this._task = task;
-    this._getDependencies = getDependencies;
+    this._args = args || (((_x: unknown) => []) as unknown as ArgsFunction<T>);
     this.taskComplete = new Promise((res, rej) => {
       this._resolveTaskComplete = res;
       this._rejectTaskComplete = rej;
     });
   }
 
   hostUpdated() {
-    this._completeTask();
+    this.runTask();
   }
 
-  private async _completeTask() {
-    const deps = this._getDependencies();
-    if (this._isDirty(deps)) {
+  protected async runTask() {
```

Comment:
Can this be made public and named `run()`? Or is there another public method to manually run a task?

Reply:
Done.